### ISSUE TITLE
Logging for infer.hypotest

### ISSUE THREAD

**Issue by alexander-held** [scikit-hep/pyhf#1374]
# Description

Calls of `infer.hypotest`, in particular with `return_expected=True`, will perform multiple fits without informing the user about the progress via log messages. For cases where each fit takes more than a few seconds it would be convenient to see logging messages that inform about the progress. This is useful particularly for interactive usage.

## Is your feature request related to a problem? Please describe.
It is difficult to estimate the progress of `infer.hypotest` calls when not having timed individual MLE fits.

### Describe the solution you'd like
Logging messages (maybe `DEBUG` level if you think `INFO` is too noisy?) after each individual fit has been performed.

### Describe alternatives you've considered
Cannot think of an alternative providing the same kind of feature.

# Relevant Issues and Pull Requests
none

# Additional context
none

**Comment by lukasheinrich** [comment]
one thing i'm curious how to do is how to not have to turn on logging for all of pyhf while wanting feedback for partiuclar calls

generally I think it's discouraged for library code to control log-levels but the user could do something like this

```
import logging

#get logger for hypotest and all its submodules
log = logging.getLogger('pyhf.infer.hypotest')

#set level to INFO or above
log.setLevel(logging.INFO)

pyhf.infer.hypotest(....)

#put back logging to only print equal or above WARNING
log.setLevel(logging.WARNING)
```

but that might be to verbose .. ew could maybe provide a utility that performs the bracket above?

```
with pyhf.logging('pyhf.infer.hypotest', "INFO"):
      pyhf.infer.hypotest(....)
```

thoughts @matthewfeickert @kratsg ?

**Comment by kratsg** [comment]
```python
import logging
from contextlib import contextmanager

@contextmanager
def log(level):
    logger = logging.getLogger()
    current_level = logger.getEffectiveLevel()
    logger.setLevel(level)
    try:
        yield
    finally:
        logger.setLevel(current_level)

with log(logging.DEBUG):
    some_function()
```

would do it

**Comment by lukasheinrich** [comment]
yeah thats what I thought of.. the quesstion is do we provide this  or let the user figure it out (right now we don't log anythinng in parrticular.. so we'd need to add log messages)

**Comment by lukasheinrich** [comment]
yeah .. one thing is what if we want to emit logging mesasges from a  tight loop i'd rarther have an option to wrap the function .. but if no logging is requested not pay the price of logging

**Comment by lukasheinrich** [comment]
one thing that might be interesting is whether we could proviide an ETA given the time needed to eval the likelihood once (which is easily meassured) and the `maxiter` settings of the optimizers)
